tools/autograd/gen_variable_type.py

```diff
@@ -1003,9 +1034,10 @@ def find_args_with_derivatives(
         # we shouldn't error out though (out= ops for autograd just redispatch)
         and len(f.func.returns) > 0
     ):
-        raise RuntimeError(
-            f"ERROR: derivative ignored for {name} -- specified an autograd function without derivative"
-        )
+        if not info.func.func.name.name.base.startswith("_foreach_"):
```

Comment:
This is a temporary condition till I figure out how I can codegen manually defined foreach derivatives